tools/xtce_generation/telemetry_generator.py

```diff
@@ -26,298 +64,280 @@ class TelemetryGenerator:
 
     """
 
-    def __init__(self, packet_name, path_to_excel_file, apid, pkt=None):
-        self.packet_name = packet_name
-        self.apid = apid
-        self.source_link = "http://www.omg.org/space/xtce"
-
-        if pkt is None:
-            # Read excel sheet
-            xls = pd.ExcelFile(path_to_excel_file)
-            self.pkt = xls.parse(packet_name)
-        else:
-            self.pkt = pkt
-
-    def create_telemetry_xml(self):
-        """
-        Create an XML representation of telemetry data based on input parameters.
-
-        Returns:
-        - root: The root element of the generated XML tree.
-        - parameter_type_set: The ParameterTypeSet element.
-        - parameter_set: The ParameterSet element.
-        - telemetry_metadata: The TelemetryMetaData element.
-        """
+    def __init__(self, path_to_excel_file):
+        # Read in all sheets from the excel file
+        self.sheets = pd.read_excel(path_to_excel_file, sheet_name=None)
+        # Set up the packet mapping from packetName to Apid
+        packet_sheet = self.sheets["Packets"]
+        self._packet_mapping = packet_sheet.set_index("packetName")["apId"].to_dict()
+
+        # Create the XML containers that will be populated later
+        self._setup_xml_containers()
+        # Add the CCSDS Header information to the containers
+        self._setup_ccsds_header()
+        # Create the sequence containers (also adding parameters within)
+        self._create_container_sets()
+
+    def _setup_xml_containers(self):
+        """Create an XML representation of telemetry data."""
```

Comment:
can you add example of what we are creating here in doc string as preview of what's been done here? I think I follow it right now by looking at existing xtce file but would be nice to see it here as well. Eg. seems like we are creating main structure of xtce here.

```
<xtce:SpaceSystem xmlns:xtce="http://www.omg.org/space/xtce" name="H45_SCI_DE">
	<xtce:Header date="2023-12" version="1.0" author="IMAP SDC" />
	<xtce:TelemetryMetaData>
		<xtce:ParameterTypeSet>
		</xtce:ParameterTypeSet>
		<xtce:ParameterSet>
		</xtce:ParameterSet>
		<xtce:ContainerSet>
		</xtce:ContainerSet>
	</xtce:TelemetryMetaData>
</xtce:SpaceSystem>
```